universe/base.go

```diff
@@ -126,6 +127,71 @@ func (a *Archive) RootNodes(ctx context.Context,
 	return a.cfg.Multiverse.RootNodes(ctx, withAmountsById)
 }
 
+// MultiverseRoot returns the root node of the multiverse for the specified
+// proof type. If the given list of universe IDs is non-empty, then the root
+// will be calculated just for those universes.
+func (a *Archive) MultiverseRoot(ctx context.Context, proofType ProofType,
+	filterByIDs []Identifier) (mssmt.Node, error) {
+
+	log.Debugf("Fetching multiverse root for proof type: %v", proofType)
+
+	leaveIDs, err := a.cfg.Multiverse.FetchMultiverseLeaves(
+		ctx, nil, nil, proofType,
```

Comment:
Rather than fetch everything all the time, we can instead pass the identifier in (if they exist) and use db filtering for it.